zebra-consensus/src/transaction.rs

```diff
@@ -518,6 +423,114 @@ where
         checks
     }
 
+    /// Verifies a transaction's Sapling shielded data.
+    async fn verify_sapling_shielded_data(
+        sapling_shielded_data: &Option<sapling::ShieldedData<sapling::PerSpendAnchor>>,
```

Comment:
Consensus rule: we need to verify sapling in V5 transactions

To make this change, we'll need to handle the `AnchorVariant` generic.

We can either:
* remove the `sapling_shielded_data` argument, and replace it with the results of the methods we call on it
  (they all return concrete types)
* make the method generic over the `AnchorVariant` trait
  (I have had trouble getting this to work in the past)

A large generic async method will also generate twice as much code, but that is unlikely to affect performance.

Reply:
Feel free to leave all these changes for the next PR, or make the method changes in this PR and the V5 changes in the next ones.

Reply:
I have the second option already implemented to be opened in a separate PR 😅 My plan is to follow this PR with the `AsyncChecks` refactor PR and then with the PR that actually has V5 support :+1: